src/libcrun/handlers/krun.c

```diff
@@ -360,6 +361,20 @@ libkrun_exec (void *cookie, libcrun_container_t *container, const char *pathname
       ret = krun_set_vm_config (ctx_id, num_vcpus, ram_mib);
       if (UNLIKELY (ret < 0))
         error (EXIT_FAILURE, -ret, "could not set krun vm configuration");
+
+      krun_set_gpu_options = dlsym (handle, "krun_set_gpu_options");
+      if (krun_set_gpu_options != NULL && access ("/dev/dri", F_OK) == 0)
+        {
+
+          uint32_t virgl_flags = VIRGLRENDERER_USE_EGL |           /* virtio-gpu compatible interface */
```

Comment:
Since the goal is offloading compute tasks, we need to rune this flags a bit. Please drop `VIRGLRENDERER_USE_EGL` (this is what's causing the `libEGL.so.1` errors) and add `VIRGLRENDERER_NO_VIRGL`.

As a side note, we could enable a different set of flags to allow running graphical apps that interact with the host's compositor from within the container. But, except for the most simpler ones, graphical apps tend to depend on various other host services, so I think it's better leaving this use case to specialized tools such as muvm.

Reply:
Fixed! Thank you for the clarification.